src/main/java/com/salesforce/storm/spout/dynamic/kafka/Consumer.java

```diff
@@ -329,6 +363,30 @@ public Record nextRecord() {
     }
 
     /**
+     * Reports kafka consumer statistics at an interval.
+     */
+    private void reportStatus() {
+        // If we've reported status more recently than 30 seconds we skip over this.
+        if (lastReportedStatusMillis != null && TimeUnit.MILLISECONDS.toSeconds(clock.millis() - lastReportedStatusMillis) < 30) {
+            return;
+        }
+
+        final Map<TopicPartition, Long> topicPartitions = getKafkaConsumer().endOffsets(
+            getKafkaConsumer().assignment()
+        );
+
+        for (final Map.Entry<TopicPartition, Long> topicPartition : topicPartitions.entrySet()) {
+            metricsRecorder.assignValue(
+                getClass(),
+                "endOffset.topic." + topicPartition.getKey().topic() + ".partition." + topicPartition.getKey().partition(),
```

Comment:
the ".topic." partition in the key might not be needed

Reply:
I suppose it's implied, but if the same topology has two instances of the same spout wouldn't that cause this to overlap?